tasks/main.yml

```diff
@@ -62,17 +62,30 @@
 - name: Install apparmor
   apt: pkg=apparmor state=present
 
+- name: Write /etc/default/docker config file
+  template: src=etc_default_docker.j2 dest=/etc/default/docker
+  register: r_etc_default_docker
+
 - name: Install LXC Docker
   include: lxc-docker.yml
-  when: "{{ docker_version == '1.5.0' }}"
+  when: docker_version == '1.5.0'
 
 - name: Install Docker Engine
   include: docker-engine.yml
-  when: "{{ docker_version != '1.5.0' }}"
+  when: docker_version != '1.5.0'
+
+# if we changed the config, but didn't reinstall docker, then restart it
+- name: Restart docker on config change
+  service: name=docker state=restarted
+  when: r_etc_default_docker.changed and not r_docker_package_install.changed
 
 - name: Ensure that Docker is running
   service: name="docker" state=started
 
+- name: reread docker facts
+  setup: filter=ansible_docker0
+  when: r_etc_default_docker|changed or r_docker_package_install|changed
```

Comment:
Why the `|changed` filter here, but `.changed` on the same variables 7 lines earlier?